maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -553,9 +553,8 @@ else if ( cliRequest.commandLine.hasOption( CLIManager.BATCH_MODE )
             logFile = resolveFile( logFile, cliRequest.workingDirectory );
 
             // redirect stdout and stderr to file
-            try
+            try ( PrintStream ps = new PrintStream( new FileOutputStream( logFile ) ) )
```

Comment:
PrintStream `ps` should not be closed here because it is passed to `System.setOut` and `System.setErr`. It will be automatically closed when `try-with-resource` block ends.

Reply:
I I think you are right. This could leave to side effects. Folks, other opinions?